test/test_transforms.py

```diff
@@ -1997,5 +1832,175 @@ def test_random_order():
+    assert (-10 < angle < 10)
+
+    # Checking if RandomRotation can be printed as string
+    t.__repr__()
+
+    # assert deprecation warning and non-BC
+    with pytest.warns(UserWarning, match=r"Argument resample is deprecated and will be removed"):
+        t = transforms.RandomRotation((-10, 10), resample=2)
+        assert t.interpolation == transforms.InterpolationMode.BILINEAR
+
+    # assert changed type warning
+    with pytest.warns(UserWarning, match=r"Argument interpolation should be of type InterpolationMode"):
+        t = transforms.RandomRotation((-10, 10), interpolation=2)
+        assert t.interpolation == transforms.InterpolationMode.BILINEAR
+
+
+def test_randomperspective():
+    for _ in range(10):
+        height = random.randint(24, 32) * 2
+        width = random.randint(24, 32) * 2
+        img = torch.ones(3, height, width)
+        to_pil_image = transforms.ToPILImage()
+        img = to_pil_image(img)
+        perp = transforms.RandomPerspective()
+        startpoints, endpoints = perp.get_params(width, height, 0.5)
+        tr_img = F.perspective(img, startpoints, endpoints)
+        tr_img2 = F.to_tensor(F.perspective(tr_img, endpoints, startpoints))
+        tr_img = F.to_tensor(tr_img)
+        assert img.size[0] == width
+        assert img.size[1] == height
+        assert (torch.nn.functional.mse_loss(tr_img, F.to_tensor(img)) + 0.3 >
+                torch.nn.functional.mse_loss(tr_img2, F.to_tensor(img)))
+
+
+def test_randomperspective_fill():
+
+    # assert fill being either a Sequence or a Number
+    with pytest.raises(TypeError):
+        transforms.RandomPerspective(fill={})
+
+    t = transforms.RandomPerspective(fill=None)
+    assert t.fill == 0
+
+    height = 100
+    width = 100
+    img = torch.ones(3, height, width)
+    to_pil_image = transforms.ToPILImage()
+    img = to_pil_image(img)
+
+    modes = ("L", "RGB", "F")
+    nums_bands = [len(mode) for mode in modes]
+    fill = 127
+
+    for mode, num_bands in zip(modes, nums_bands):
+        img_conv = img.convert(mode)
+        perspective = transforms.RandomPerspective(p=1, fill=fill)
+        tr_img = perspective(img_conv)
+        pixel = tr_img.getpixel((0, 0))
+
+        if not isinstance(pixel, tuple):
+            pixel = (pixel,)
+        assert pixel == tuple([fill] * num_bands)
+
+    for mode, num_bands in zip(modes, nums_bands):
+        img_conv = img.convert(mode)
+        startpoints, endpoints = transforms.RandomPerspective.get_params(width, height, 0.5)
+        tr_img = F.perspective(img_conv, startpoints, endpoints, fill=fill)
+        pixel = tr_img.getpixel((0, 0))
+
+        if not isinstance(pixel, tuple):
+            pixel = (pixel,)
+        assert pixel == tuple([fill] * num_bands)
+
+        for wrong_num_bands in set(nums_bands) - {num_bands}:
+            with pytest.raises(ValueError):
+                F.perspective(img_conv, startpoints, endpoints, fill=tuple([fill] * wrong_num_bands))
```

Comment:
This hinders the use of something like following
```
pytest.mark.parametrize('band', ("L", "RGB", "F"))
num_bands = len(band)
```
Since `nums_bands` is also required here.
Let me know if you have any comments here. Thanks in advance :)